optimum/habana/transformers/generation/utils.py

```diff
@@ -2029,7 +2046,16 @@ def _contrastive_search(
             ):
                 # Pad the returned pask key values tensors from prefill phase forward run to maximum length
                 # before starting the decode phase.
-                self._pad_past_key_values(model_kwargs)
+
+                is_mqa_model = self.config.model_type == 'gpt_bigcode' and self.config.multi_query
```

Comment:
hi @jiminha I only tested this code for gpt_bigcode models, so I don't know if it will fit all MQA models, that's why I added model check here. Also ```falcon``` in OptimumHabana is implemented with ```use_cache```, not ```internal_bucketing```, so it is even not possible to have a try. So I believe it is safer to have model check here and it will be very easy to extend or remove in future

Reply:
Got it understand.